docs/hooks/appointment-booking.md

```diff
@@ -0,0 +1,237 @@
+
+## Context
+
+The Patient whose appointment is/appointments are bing booked as well as the draft Appointment records.
+
+Field | Optionality | Prefetch Token | Type | Description
+----- | -------- | ---- | ---- | ----
+`patientId` | REQUIRED | Yes | *string* | The FHIR `Patient.id` of Patient appointment(s) is/are for
+`encounterId` | OPTIONAL | Yes | *string* | The FHIR `Encounter.id` of Encounter where booking was initiated
+`appointments` | REQUIRED | No | *object* | DSTU2/STU3/R4 - FHIR Bundle of Appointments in 'proposed' state
+
+
+### Examples
+
+```json
+"context":{
+  "patientId" : "1288992",
+  "appointments" : [
+    {
+      "resourceType": "Appointment",
+      "id": "apt1",
+      "status": "proposed",
+      "serviceType": [
+        {
+          "coding": [
+            {
+              "code": "183",
+              "display": "Sleep Medicine"
+            }
+          ]
+        }
+      ],
+      "appointmentType": {
+        "coding": [
+          {
+            "system": "http://hl7.org/fhir/v2/0276",
+            "code": "FOLLOWUP",
+            "display": "A follow up visit from a previous appointment"
+          }
+        ]
+      },
+      "reason": {
+        "coding": {
+          "system": "",
+          "code": "1023001",
+          "display": "Apnea"
+        }
+      },
+      "description": "CPAP adjustments",
+      "start": "2019-08-10T09:00:00-06:00",
+      "end": "2019-08-10T09:10:00:00-06:00",
+      "created": "2019-08-01",
+      "participant": [
+        {
+          "actor": {
+            "reference": "Patient/example",
+            "display": "Peter James Chalmers"
+          },
+          "required": "required",
+          "status": "tentative"
+        },
+        {
+          "actor": {
+            "reference": "Practitioner/example",
+            "display": "Dr Adam Careful"
+          },
+          "required": "required",
+          "status": "accepted"
+        }
+      ]
+    },
+    {
+      "resourceType": "Appointment",
+      "id": "apt1",
+      "status": "proposed",
+      "appointmentType": {
+        "coding": [
+          {
+            "system": "http://hl7.org/fhir/v2/0276",
+            "code": "CHECKUP",
+            "display": "A routine check-up, such as an annual physical"
+          }
+        ]
+      },
+      "description": "Regular physical",
+      "start": "2020-08-01T13:00:00-06:00",
+      "end": "2020-08-01T13:30:00:00-06:00",
+      "created": "2019-08-01",
+      "participant": [
+        {
+          "actor": {
+            "reference": "Patient/example",
+            "display": "Peter James Chalmers"
+          },
+          "required": "required",
+          "status": "tentative"
+        },
+        {
+          "actor": {
+            "reference": "Practitioner/example",
+            "display": "Dr Adam Careful"
+          },
+          "required": "required",
+          "status": "accepted"
+        }
+      ]
+    }
+  ]
+}
+```
+
+```json
```

Comment:
Is this the R4 example? 

Could you please label each example with the FHIR version it implements?

Reply:
Both are STU3.  Label added